### ISSUE TITLE
[idea] new Op: Conv2DWithBiasActivation for Backends that Support Higher Level Op

### ISSUE THREAD

**Issue by cancan101** [tensorflow/tensorflow#9272]
This is a [repost of the comment here](https://github.com/tensorflow/tensorflow/pull/8673#issuecomment-290722014):

Along the lines of [`MklConv2DWithBias`](https://github.com/tensorflow/tensorflow/blob/904edee4456a61d50d5b1ffe9858a7772acc423e/tensorflow/core/ops/nn_ops.cc#L2632-L2645), I suggest defining a new primitive Op, `Conv2DWithBiasActivation` that can be used to define kernels on platforms that support a higher level primitive conv-bias-acitivation unit.

As far as the (new) fused `Conv2D` Op that I am using for my BNNS implementation, I used the [MKL Conv2D Op](https://github.com/tensorflow/tensorflow/blob/904edee4456a61d50d5b1ffe9858a7772acc423e/tensorflow/core/ops/nn_ops.cc#L2632-L2645) as a predicate. In that case Intel has defined a new Op that includes both the convolution Op and the BiasAdd Op. My Conv Op actually isn't pulling in just the activation function, rather, my Op and the MKL both have the bias add has been fused as well.

I understand that the primitive `Conv2D` op in TF does not currently handle activations or bias addition, but this op may be "too primitive". In fact cuDNN is now moving in the direction of fusing these three into one kernel (see: https://github.com/tensorflow/tensorflow/issues/8828). Perhaps, a different FusedConv2D Op should be added that can be used by not just BNNS but also cuDNN v6 and any other underlying platform implementations that can take advantage of doing all three ops together. This would be  along the lines of batch norm, which now offers a `fused` version of the Op to reduce the number of primitive Ops used.

Addendum
There are even more implementations that can benefit from fusing Conv-Bias-Activation. Others include:
* Metal Performance Shaders (https://github.com/tensorflow/tensorflow/issues/7958). Here it might be extra extra important because of 1. avoiding extra copies back and forth from GPU memory (ie using the tuned `MPSTemporary​Image` and 2. because MPS uses a very weird memory layout and ideally transposing to this only needs to be done once.
* MKL (as linked above, perhaps the MKL specific Op (`MklConv2DWithBias`) can be unified with this new Op.
* [cuDNN v6](https://github.com/tensorflow/tensorflow/issues/8828)

/CC @drpngx @petewarden @flx42 @gunan

**Comment by drpngx** [comment]
Assigning @petewarden but feel free to bounce.

**Comment by petewarden** [comment]
We're going to be working on fusing ops like this as part of TF Lite, so I don't think this change will make it into the main code base. Closing, but please reopen if you strongly feel the fused ops should exist at this level.

**Comment by cancan101** [comment]
I don't know what TF Lite is, but I guess I do think this would be valuable for TF main.
